DOCS-6477 USM updates for private beta (#20295)
DataDog/documentation: content/en/universal_service_monitoring/setup.md

---
title: Setting up Universal Service Monitoring
kind: Documentation
further_reading:
- link: "/universal_service_monitoring/"
  tag: "Documentation"
  text: "Universal Service Monitoring"
- link: "https://www.datadoghq.com/blog/universal-service-monitoring-datadog/"
  tag: "Blog"
  text: "Golden signals in seconds with Universal Service Monitoring"
---

## Supported versions and compatibility

Required Agent version

> : Universal Service Monitoring requires that the Datadog Agent installed alongside your containerized service be at least version 6.40 or 7.40, higher as noted below for some beta features.

Reviewer: ```suggestion
: Universal Service Monitoring requires that the Datadog Agent installed alongside your containerized service be at least version 6.40 or 7.40. As noted below, some beta features require higher versions.
```
I think this reads better as a new sentence

Supported Linux platforms
: Linux Kernel 4.14 and greater<br/>
CentOS or RHEL 8.0 and greater

Supported Windows platforms
: IIS on Windows 2012 R2 and greater

Supported application-layer protocols
: HTTP<br/>
HTTPS (OpenSSL)

<div class="alert alert-info">
Additional protocols and traffic encryption methods are supported in <a href="/universal_service_monitoring/additional_protocols/">private beta</a>. If you have feedback about what platforms and protocols you'd like to see supported, <a href="/help/">contact Support</a>.
</div>

## Prerequisites

- If on Linux:
    - Your service is running in a container.
    - **Beta:** For non-containerized services see the [instructions here](#non-containerized-services-on-linux).

> - If on Windows and using IIS:

Reviewer: ```suggestion
- If on Windows IIS:
```
Based on the supported versions above, IIS is required. Edited cos the sentence might imply to some that it's optional.

     - /proc/:/host/proc/:ro
     - /sys/fs/cgroup/:/host/sys/fs/cgroup:ro
     - /sys/kernel/debug:/sys/kernel/debug
     - /lib/modules:/lib/modules
     - /usr/src:/usr/src
     - /var/tmp/datadog-agent/system-probe/build:/var/tmp/datadog-agent/system-probe/build
     - /var/tmp/datadog-agent/system-probe/kernel-headers:/var/tmp/datadog-agent/system-probe/kernel-headers
     - /etc/apt:/host/etc/apt
     - /etc/yum.repos.d:/host/etc/yum.repos.d
     - /etc/zypp:/host/etc/zypp
     - /etc/pki:/host/etc/pki
     - /etc/yum/vars:/host/etc/yum/vars
     - /etc/dnf/vars:/host/etc/dnf/vars
     - /etc/rhsm:/host/etc/rhsm
    cap_add:
     - SYS_ADMIN
     - SYS_RESOURCE
     - SYS_PTRACE
     - NET_ADMIN
     - NET_BROADCAST
     - NET_RAW
     - IPC_LOCK
     - CHOWN
    security_opt:
     - apparmor:unconfined
```

For optional HTTPS support, also add:

```yaml
services:
  ...
  datadog:
    ...
    environment:
     - HOST_ROOT: '/host/root'
    volumes:
     - /:/host/root:ro
```

{{% /tab %}}
{{% tab "Docker Swarm" %}}

As `Docker Swarm` does not yet support the changing of `security_opt`, the operating system
must not have a running `apparmor` instance.

> If the operating system does not have a running `apparmor` instance, then use the same `docker-compose.yml` file from the `Docker-Compose` [section][1] beside the field `security_opt`.

Reviewer: ```suggestion
If the operating system does not have a running `apparmor` instance, use the same `docker-compose.yml` file from the `Docker-Compose` [section][1] beside the field `security_opt`.
```

> [1]: /universal_service_monitoring/?tab=dockercompose#enabling-universal-service-monitoring

Reviewer: ```suggestion
[1]: /universal_service_monitoring/setup/?tab=dockercompose#enabling-universal-service-monitoring
```
Not sure if you can just use `?tab=dockercompose#enabling-universal-service-monitoring`

2. Enable ENI and EC2 metric collection.
3. Add the following tags to each load balancer:
   ```conf
   ENV=<env>
   SERVICE=<service>
   ```

[1]: /containers/amazon_ecs/?tab=awscli#run-the-agent-as-a-daemon-service
[2]: /integrations/amazon_web_services/
{{% /tab %}}

{{% tab "Windows" %}}

**For services running on IIS:**

1. Install the [Datadog Agent][1] (version 6.41 or 7.41 and later) with the network kernel device driver component enabled.
   For Agent version 7.44 or earlier, you must pass `ADDLOCAL="MainApplication,NPM"` to the `msiexec` command during installation, or select **Network Performance Monitoring** when running the Agent installation through the GUI.

2. Edit `C:\ProgramData\Datadog\system-probe.yaml` to set the enabled flag to `true`:

   ```yaml
   service_monitoring_config:
     enabled: true
   ```
[1]: /agent/basic_agent_usage/windows/?tab=commandline
{{% /tab %}}

{{< /tabs >}}

### Non-containerized services on Linux

<div class="alert alert-info">
Universal Service Monitoring is available in <strong>beta</strong> to monitor services running bare-metal on Linux virtual machines.
</div>

> Requires agent of version 7.42+.

Reviewer: ```suggestion
Requires Agent version 7.42 or greater.
```
